ledger/participant-integration-api/src/main/scala/platform/index/InMemoryStateUpdater.scala

```diff
@@ -41,11 +52,12 @@ final class InMemoryStateUpdater(
       .filter(_._1.nonEmpty)
       .mapAsync(prepareUpdatesParallelism) { case (batch, lastEventSequentialId) =>
         Future {
-          val transactionsAcceptedBatch =
-            batch.collect { case (offset, u: TransactionAccepted) =>
-              updateToTransactionAccepted(offset, u)
+          val updatesBatch =
+            batch.collect {
+              case (offset, u: Update.TransactionAccepted) => convertTransactionAccepted(offset, u)
+              case (offset, u: Update.CommandRejected) => convertTransactionRejected(offset, u)
             }
-          transactionsAcceptedBatch -> (batch.last._1 -> lastEventSequentialId)
+          updatesBatch -> (batch.last._1 -> lastEventSequentialId)
```

Comment:
If the intention is to create a short lived tuple to pass to the next processing stage how about:
`(updatesBatch, batch.last._1, lastEventSequentialId)`
then match with 
`.mapAsync { case (updates, lastOffset, lastEventSequentialId) =>`

Reply:
sure, why not